prime-router/src/main/kotlin/cli/helpers/HL7DiffHelper.kt

```diff
@@ -245,7 +251,12 @@ class HL7DiffHelper {
                 }
             }
 
-            input.javaClass != output.javaClass -> {
+            // heuristic check to avoid false positive:
+            // either input vs output class identical or
+            // both are ca.uhn.hl7v2.model.AbstractPrimitive
+            // and since value check done by above value compare logic,
+            // input vs output are type compatible
```

Comment:
(nit, optional) I think if we want to explain what are considered compatible types, this comment is a better fit at the `typeCompatible` function definition, and we can take it out here. Don't feel too strongly about it, but it seems unnecessary to have a multi-line comment at this point in the code.